src/hermes/model/context_manager.py

```diff
@@ -54,11 +54,22 @@ def __init__(self, project_dir: pathlib.Path = pathlib.Path.cwd()):
     def prepare_step(self, step: str, *depends: str) -> None:
         self._current_step.append(step)
 
+    '''
+    FIXME: Implement this to fix #373
+
+    def finalize_step(self, step: str) -> None:
+        current_step = self._current_step[-1]
+        if current_step != step:
+            raise ValueError(f"Cannot end step {step} while in {self._current_step[-1]}.")
+        self._current_step.pop()
```

Comment:
Why do you put the "fix" into the comment instead of putting it into the implementation?

The however, the first line will still throw an IndexError for empty lists.

Reply:
Documented in #373.